sdk/servicebus/service-bus/src/core/batchingReceiver.ts

```diff
@@ -325,15 +325,23 @@ export class BatchingReceiverLite {
     const brokeredMessages: ServiceBusMessageImpl[] = [];
     const loggingPrefix = `[${receiver.connection.id}|r:${receiver.name}]`;
 
-    return new Promise<ServiceBusMessageImpl[]>((resolve, reject) => {
+    return new Promise<ServiceBusMessageImpl[]>((origResolve, origReject) => {
       let totalWaitTimer: NodeJS.Timer | undefined;
 
       // eslint-disable-next-line prefer-const
       let cleanupBeforeResolveOrReject: () => void;
 
-      const onError: OnAmqpEvent = (context: EventContext) => {
+      const reject = (err: Error | AmqpError) => { 
         cleanupBeforeResolveOrReject();
+        origReject(err);
```

Comment:
As if being rejected once wasn't bad enough...

Reply:
At least rejection is immediate!